docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,251 @@ Note that:
+     {
+       "path": "/path/to/file/foo.conf"
+       "type": "foo.conf"
+     },
+     {
+       "path": "/another/path/to/file/bar.conf"     
+     }
+   ]
+}
+```
+
+Whenever the `c8y_configuration_plugin` receivces that MQTT message it stores all contained information to the child-devices individual TOML file in `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml`. When the file already exists it will be replaced with the new content.
+
+Each time a child-device's TOML is modified (e.g. due to an incoming MQTT message, an incoming new configuration snapshot from the cloud, or a modification by a local process) the `c8y_configuration_plugin` takes care to define all necessary capabilities to the coresponding cloud's child-device twin. These are:
+  - declaring _supported operations_ for configuration management: `c8y_UploadConfigFile` and `c8y_DownloadConfigFile`
+  - declaring provided _configuration types_
+
+**Declaring 'supported operations'**
+
+To declare supported operations the `c8y_configuration_plugin` uses thin-edge's _Supported Operations API_. Therefore the `c8y_configuration_plugin` creates for each child-device two files under `/etc/tedge/operations/c8y/<childid>`.
+
+Example, for child-device with childid `child1`:
+
+```
+/etc/tedge/operations/c8y/child1/c8y_UploadConfigFile
+/etc/tedge/operations/c8y/child1/c8y_DownloadConfigFile
+```
+
+As soon as those files are created, thin-edge's _Supported Operations API_ takes care to send according supported operation declarations to cloud's child-device twins (see [documentation Supported Operations](../tutorials/supported_operations.md#supported-operations-for-child-devices)).
+
+**Declaring 'provided configuration types'**
+
+For all configuration `types` provided by the child-device, the plugin sends an MQTT message to C8Y. Thereby all `types` of the child device's `c8y-config-plugin.toml` with the added type `c8y-config-plugin` to represent the config list TOML file itself will be combined in one single message:
+  - topic: `c8y/s/us/<childid>`
+  - payload: `119,c8y-config-plugin,<type 1>,<type 2>,<type 3>,...`<br/>
+    Example: `119,c8y-config-plugin,foo.conf,bar.conf`
+
+Note that the `c8y_configuration_plugin` does **not** create any child-device twin in the cloud. Instead, the cloud child-device twins must be created upfront.
+
+
+## Fetching/Pushing configuration file from/to the external device
+
+To fetch/push configuration files to/from external devices, the `c8y_configuration_plugin` makes use of file transfer HTTP APIs of the `tedge_agent`.
+  
+The HTTP file transfer feature of the `tedge_agent` provides the service to transfer files from external devices to the local filesystem of the thin-edge device's, and vice versa.
+
+  * `tedge_agent` serves PUT, GET and DELETE requests for temporary file upload, download and removal
+    * a temporary file can be uploaded with an HTTP PUT request to `http://<thin-edge IP address>/tedge/tmpfiles`,
```

Comment:
```suggestion
    * a temporary file can be uploaded with an HTTP PUT request to `http://<thin-edge IP address>/tedge/configuration_snapshot/{file-id}`,
```
Update this URL in all the references below.

Reply:
I think the URL you suggested here is not aligned with the one of the prototype's doc. There it is `/tedge/config_snapshot/{file-id}`, instead of `/tedge/configuration_snapshot/{file-id}`.

Is changed all HTTP URLs to the one of the prototype's doc.